fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexPrefetchRangeKeyValueCursor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * IndexRangeKeyValueCursor.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.KeySelector;
+import com.apple.foundationdb.KeyValue;
+import com.apple.foundationdb.MappedKeyValue;
+import com.apple.foundationdb.ReadTransaction;
+import com.apple.foundationdb.StreamingMode;
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.async.AsyncIterator;
+import com.apple.foundationdb.record.cursors.CursorLimitManager;
+import com.apple.foundationdb.subspace.Subspace;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A {@link KeyValueCursor} that scans an index using the IndexPrefetch operation.
+ * This subclass of {@link KeyValueCursor} uses a different scan operation ({@link com.apple.foundationdb.Transaction#getMappedRange})
+ * and therefore the actual returned types of the scanned range are different too - they need to be parsed into records
+ * rather than Index entries. The returned values are MappedKeyValues and contain all splits of
+ * the record for a particular index entry.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class IndexPrefetchRangeKeyValueCursor extends KeyValueCursor {
```

Comment:
It kind of seems like rather than having `IndexPrefetchRangeKeyValueCursor` extend `KeyValueCursor` that both should extend a new abstract class called something like `BaseKeyValueCursor<K extends KeyValue> implements BaseCursor<K>`.

I also don't think it's necessarily critical, and if we wanted to do that, but in a later PR, that might be fine

Reply:
Created an issue